publish/npm/install.js

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env node
+
+const fs = require("fs");
+const path = require("path");
+const https = require("https");
+const crypto = require("crypto");
+const { execSync } = require("child_process");
+const {
+  BINARY_NAME,
+  REPO_OWNER,
+  REPO_NAME,
+  GITHUB_RELEASES_BASE,
+  BINARY_PATTERNS,
+} = require("./config");
+
+// Read version from package.json with strict validation
+function getValidatedVersion() {
+  try {
+    const packageJson = JSON.parse(
+      fs.readFileSync(path.join(__dirname, "package.json"), "utf8"),
+    );
+
+    const version = packageJson.version;
+
+    // Strict validation: must be valid semver (x.y.z)
+    if (!/^\d+\.\d+\.\d+$/.test(version)) {
+      throw new Error(`Invalid version format: ${version}`);
+    }
+
+    return `v${version}`;
+  } catch (error) {
+    throw new Error(`Failed to read valid version: ${error.message}`);
+  }
+}
+
+const RELEASE_VERSION = getValidatedVersion();
+const BASE_URL = `${GITHUB_RELEASES_BASE}/${RELEASE_VERSION}`;
+
+// Platform-specific binary URLs (constructed from config)
+const BINARY_URLS = {};
+Object.keys(BINARY_PATTERNS).forEach((platform) => {
+  BINARY_URLS[platform] = `${BASE_URL}/${BINARY_PATTERNS[platform]}`;
+});
+
+const CHECKSUMS_URL = `${BASE_URL}/checksums.txt`;
+
+function getPlatformKey() {
+  const platform = process.platform;
+  const arch = process.arch;
+  return `${platform}-${arch}`;
+}
+
+function downloadFile(url, dest) {
+  return new Promise((resolve, reject) => {
+    const file = fs.createWriteStream(dest);
+
+    https
+      .get(url, (response) => {
+        if (response.statusCode === 302 || response.statusCode === 301) {
+          return downloadFile(response.headers.location, dest)
```

Comment:
There should be limit to handling redirects. Else a misconfigured web server will continuously send redirects and lead to stack exhaustion here. May be you can pass a counter to `downloadFile` which fails after handling a given number of redirects